sdlib/d/gc/hpd.d

```diff
@@ -111,6 +111,44 @@ public:
 		return bestIndex;
 	}
 
+	bool reserveAt(uint index, uint pages) {
+		assert(pages > 0 && pages <= PageCount, "Invalid number of pages!");
+		assert(index <= PageCount - pages, "Invalid index!");
+
+		if (pages > longestFreeRange) {
+			return false;
+		}
+
+		uint freePos, freeLength;
+		allocatedPages.nextFreeRange(index, freePos, freeLength);
+
+		if ((freePos != index) || (freeLength < pages)) {
+			return false;
+		}
+
+		allocatedPages.setRange(index, pages);
+		usedCount += pages;
+
+		// If allocated from the longest free range, compute the new one:
+		if (freeLength == longestFreeRange) {
+			auto longestFree = longestFreeRange - pages;
+			uint i = 0;
+			while (
+				i < PageCount
```

Comment:
If you feel fancy, `PageCount - longestFree` with a lot of tests. if not, then just add a comment.